wayrs-scanner/src/lib.rs

```diff
@@ -403,14 +403,25 @@ fn gen_pub_fn(
     generics: &[TokenStream],
     args: &[TokenStream],
     ret_ty: TokenStream,
+    where_: Option<TokenStream>,
     body: TokenStream,
 ) -> TokenStream {
     let name = make_ident(name);
-    quote! {
-        #attrs
-        #[allow(clippy::too_many_arguments)]
-        pub fn #name<#(#generics),*>(#(#args),*) -> #ret_ty {
-            #body
+    if let Some(w) = where_ {
+        quote! {
+            #attrs
+            #[allow(clippy::too_many_arguments)]
+            pub fn #name<#(#generics),*>(#(#args),*) -> #ret_ty #w {
+                #body
+            }
+        }
+    } else {
+        quote! {
+            #attrs
+            #[allow(clippy::too_many_arguments)]
+            pub fn #name<#(#generics),*>(#(#args),*) -> #ret_ty {
+                #body
+            }
```

Comment:
This is not necessary: `Option<T>` can be [quoted directly](https://docs.rs/quote/latest/quote/trait.ToTokens.html#impl-ToTokens-for-Option%3CT%3E)

Reply:
Nice! Fixed.